Update Util.writeFile

alif opened the pull request:
Ensure it handles undefined data gracefully
This change will be needed when we convert to Node.js v16.
Relevant conversation: https://github.com/superhuman/desktop/pull/19277

Set the function to async
- This wasn't being handled by the callers of the function particularly well

Set function Util.readFileAsync to async

alif commented on lib/util.js:
```diff
@@ -88,16 +88,20 @@ Util.readFile = function (p, encoding) {
 
-Util.writeFile = function (p, data) {
+Util.writeFile = async function (p, data) {
   Util.ensureDir(path.dirname(p))
```
Presumably we'd want to ensure the directory even if the data is undefined, but this seems unclear to me.

Timothee reviewed (approved):
This looks good overall. I think we should tackle the likely cause of the error (`chrome.js`) one way or another.

You'll also need to update the version in `package.json` and add a `CHANGELOG` item.

Hehk commented on lib/util.js:
```diff
@@ -88,16 +88,20 @@ Util.readFile = function (p, encoding) {
 }
 
-Util.writeFile = function (p, data) {
+Util.writeFile = async function (p, data) {
```
I am in favor of having a '' default. I don't exactly trust that chrome launcher will always give a proper id and would rather not error out in that case

alif commented:
@Hehk @Timothee I made some significant changes and could use another look over.
Updated the `chrome-remote-interface` library and added some catching.
The uncaught errors no longer appear in logs.

alif commented on lib/chrome.js:
```diff
@@ -397,6 +398,7 @@ class ChromeTab {
     let isToGateway = gatewayUrl && request.url.indexOf(gatewayUrl) >= 0
     if (!this.manifest || !isToGateway) {
       return this.cdp.Fetch.continueRequest({ requestId })
+          .catch(err => { console.error('Non-fatal error from cdp.Fetch.continueRequest', err) })
```
I was able to determine that this call specifically is related to https://github.com/superhuman/desktop/pull/19277#issuecomment-1111340350
and the uncaught exceptions.

I tested Node v16 with these changes and the test suite runs locally now.

Timothee reviewed (approved):
I haven't tested it but the code looks reasonable to me.

alif commented:
Here's a testrun with [Node v16](https://superhuman.semaphoreci.com/workflows/b3266286-d169-4236-a6aa-28cec102f924?pipeline_id=90d57187-5e99-4342-968e-2a3f0f9a3dce)

Hehk reviewed (approved):
LGTM
